### ISSUE TITLE
[Feature Request] DoH advanced controls and Fallback to local DNS

### ISSUE THREAD

**Issue by brokoler** [brave/brave-browser#35791]
## Description 
Brave should give more control over DNS over HTTPS (DoH).
Right now it can be enabled, disabled and a specific DoH server can be set.
**Following features are missing:**
- Setting to choose Fallback to local DNS if Domain can't be resolved by DoH server without a warning (Similar to "Increased Protection" from Firefox)
- Setting to force only DoH server with a warning when Fallback to local DNS happens (Similar to "Max Protection" from Firefox)
- Exclusion of specific domains from DoH

This feature would allow to use a DoH resolver whenever it is possible and to have a local fallback when it is needed. At the moment someone has to write Localhost entries to reach internal Domains for each system, which is not intuitive.

Firefox added more granular control a few months ago, Brave should adopt something similar:

![image](https://github.com/brave/brave-browser/assets/57031979/9b3f992c-33c7-4ad9-8998-6aec25070d45)

**Comment by diracdeltas** [comment]
cc @arthuredelstein

**Comment by apertureless** [comment]
Yes please! The current implementation is not useable for developers. 

I tried to create a separate profile in brave, but brave does not save dns settings per profile .

**Comment by theMatthewT** [comment]
+1 for this feature.

I use secure DNS in Brave at work, but some company-specific web applications stop working because they rely on the corporate DNS server. Having a fallback to the system DNS when secure DNS cannot resolve a domain would be very useful.
